components/Common/DarkModeToggle/__tests__/index.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import DarkModeToggle from '../index';
+
+let mockCurrentTheme = '';
+
+const mockToggleTheme = jest.fn().mockImplementation(() => {
+  mockCurrentTheme = mockCurrentTheme === 'dark' ? 'light' : 'dark';
+});
+
+// // mock dark mode module for controlling dark mode HOC behavior
+jest.mock('next-themes', () => ({
+  useTheme: () => {
+    return { theme: mockCurrentTheme, setTheme: mockToggleTheme };
+  },
+}));
+
+describe('DarkModeToggle Component', () => {
+  it('render dark mode toggle', () => {
+    const { container } = render(<DarkModeToggle />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('switches dark theme to light theme', async () => {
+    mockCurrentTheme = 'dark';
+    render(<DarkModeToggle />);
+    const toggle = await screen.findByText('Toggle Dark Mode');
+    await userEvent.click(toggle);
+    expect(mockCurrentTheme).toBe('light');
+  });
+
+  it('switches light theme to dark theme', async () => {
+    mockCurrentTheme = 'light';
+    render(<DarkModeToggle />);
+    const toggle = await screen.findByText('Toggle Dark Mode');
```

Comment:
Instead of checking `findByText` (as text can change) you can check by the presence of the `light-mode-only` or `dark-mode-only` or the `aria-pressed`

Reply:
Just so you know, you shouldn't add the English messages on the IntlProvider for unit testing. See: https://github.com/nodejs/nodejs.org/pull/5279

Reply:
So the snapshots will contain the keys now instead of the en lang?

Reply:
Exactly!